app/controllers/wiki_controller.rb

```diff
@@ -72,6 +72,10 @@ def edit
     else
       @node = DrupalNode.find_wiki(params[:id])
     end
+    if @node.has_power_tag('locked') && (current_user.role != "admin" && current_user.role != "moderator")
+      flash.now[:warning] = "This page is <a href='/wiki/power-tags#Locking'>locked</a>, and only <a href='/wiki/moderators'>moderators</a> can edit it."
```

Comment:
Ah, i think the problem is here -- a `flash.now` runs **before** the redirect, but a `flash[]` runs after. If you remove the "now" it should work and pass the test!